src/Storage/Storage/File/Cmdlet/GetAzureStorageFileHandle.cs

```diff
@@ -0,0 +1,201 @@
+        public string Path { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "List handles Recursively. Only works on File Directory.")]
+        public SwitchParameter Recursive { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            CloudFileDirectory baseDirectory = null;
+            switch (this.ParameterSetName)
+            {
+                case Constants.DirectoryParameterSetName:
+                    baseDirectory = this.Directory;
+                    break;
+
+                case Constants.ShareNameParameterSetName:
+                    baseDirectory = this.BuildFileShareObjectFromName(this.ShareName).GetRootDirectoryReference();
+                    break;
+
+                case Constants.ShareParameterSetName:
+                    baseDirectory = this.Share.GetRootDirectoryReference();
+                    break;
+                case Constants.FileParameterSetName:
+                    // Don't need to set baseDirectory when input is a CloudFile
+                    break;
+
+                default:
+                    throw new PSArgumentException(string.Format(CultureInfo.InvariantCulture, "Invalid parameter set name: {0}", this.ParameterSetName));
+            }
+
+            // When not input path/File, the list handle target must be a Dir
+            bool foundAFolder = true;
+            CloudFileDirectory targetDir = baseDirectory;
+            CloudFile targetFile = null;
+            if (this.File != null)
+            {
+                targetFile = this.File;
+                foundAFolder = false;
+            }
+            else
+            {
+                if (!string.IsNullOrEmpty(this.Path))
+                {
+                    string[] subfolders = NamingUtil.ValidatePath(this.Path);
+                    targetDir = baseDirectory.GetDirectoryReferenceByPath(subfolders);
+
+                    if (!targetDir.Exists())
+                    {
+                        foundAFolder = false;
+                    }
+
+                    if (!foundAFolder)
+                    {
+                        //Get file
+                        string[] filePath = NamingUtil.ValidatePath(this.Path, true);
+                        targetFile = baseDirectory.GetFileReferenceByPath(filePath);
+
+                        this.Channel.FetchFileAttributesAsync(
+                            targetFile,
+                            null,
+                            this.RequestOptions,
+                            this.OperationContext,
+                            this.CmdletCancellationToken).ConfigureAwait(false);
+                    }
+                }
+            }
+
+            // Recursive only take effect on File Dir
+            if (!foundAFolder && Recursive.IsPresent)
+            {
+                WriteWarning("The target object of the 'Path' is an Azure File, the parameter 'Recursive' won't take effect.");
```

Comment:
WriteWarning is a bit excessive for this - it is not dangerous if the cmdlet doesn't recurse.  A Verbose message should be fine.

Reply:
Fixed.